service/history/workflow/cache/cache.go

```diff
@@ -84,13 +84,19 @@ type (
 		) (workflow.Context, ReleaseCacheFunc, error)
 	}
 
-	CacheImpl struct {
+	cacheImpl struct {
 		cache.Cache
 
+		onPut                     func(wfContext *workflow.Context)
+		onEvict                   func(wfContext *workflow.Context)
+		logger                    log.Logger
 		nonUserContextLockTimeout time.Duration
 	}
+	cacheItem struct {
```

Comment:
Do you really need it now?

Reply:
Yes, it will contain the `Finalizer` in the next PR. The Finalizer cannot be passed into the cache since it has a different lifecycle.